.github/workflows/test-code.yaml

```diff
@@ -84,6 +84,27 @@ jobs:
         with:
           check-latest-tag-only: true
 
+  gitleaks:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - name: Install gitleaks
+        shell: bash
+        run: |
+          set -e;
+          wget https://github.com/zricethezav/gitleaks/releases/download/v8.11.2/gitleaks_8.11.2_linux_x64.tar.gz;
+          echo '9e68a91906ce2b781b2bb785285cbfe5ad294e80ec3bb250fb019650a5248d15  gitleaks_8.11.2_linux_x64.tar.gz' > gitleaks_8.11.2_checksums.txt;
+          shasum -a 256 gitleaks_8.11.2_checksums.txt --check;
+          tar -xvf gitleaks_8.11.2_linux_x64.tar.gz;
```

Comment:
Do we want to put this in a `.sh` file like the other scripts to stay consistent?

Reply:
sure, on it